docs/development/local_setup.md

```diff
@@ -0,0 +1,85 @@
+# Preparing the Local Development Setup (Mac OS X)
+
+Conceptionally, the Node Controller Manager is designated to run in a container within a Pod inside a Kubernetes cluster. For development purposes, you can run the Node Controller Manager as a Go process on your local machine. This process connects to your remote cluster to manage VMs for that cluster. That means that the Node Controller Manager runs outside a Kubernetes cluster which requires providing a [Kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/) in your local filesystem and point the Node Controller Manager to it when running it (see below). 
+
+Although the following installation instructions are for Mac OS X, similar alternate commands could be found for any Linux distribution.
+
+### Installing Golang environment
+Install the latest version of Golang (at least `v1.8.3` is required) by using [Homebrew](https://brew.sh/):
+
+```bash
+$ brew install golang
+```
+
+Make sure to set your `$GOPATH` environment variable properly (conventionally, it points to `$HOME/go`).
+
+For your convenience, you can add the `bin` directory of the `$GOPATH` to your `$PATH`: `PATH=$PATH:$GOPATH/bin`, but it is not necessarily required.
+
+In order to perform linting on the Go source code, install [Golint](https://github.com/golang/lint):
+
+```bash
+$ go get -u github.com/golang/lint/golint
+```
+
+[Dep](https://github.com/golang/dep) is used for managing Golang package dependencies. Install it:
+```bash
+$ brew install dep
+```
+
+### Installing `git`
```

Comment:
I would not mention installation of git, every kubernetes developer will already have it

Reply:
Removed.
https://github.com/gardener/node-controller-manager/commit/95c68a4c3f483a5f770c7de61f12b25e5411f3c0#diff-5d5c08bc24d310cb81e8c13a38f106b9L29